drivers/include/n25q128.h

```diff
@@ -0,0 +1,206 @@
+        uint8_t send    : 1;   /*< Set, when the memory is receiving. */
+        uint8_t recv    : 1;   /*< Set, when the memory is sending. */
+        uint8_t unused  : 4;   /*< Not used bits. */
+    } enabled;
+} n25q128_opt_t;
+
+/*
+ * @brief   Type definiton of a instruction for a N25Q128.
+ */
+typedef struct {
+    uint8_t code;       /*< The code of the instruction. */
+    int32_t addr;       /*< The address to read or write data from. */
+} n25q128_cmd_t;
+
+/*
+ * @brief   Type definition for the data of a N25Q128.
+ */
+typedef struct {
+    uint8_t *buf;       /*< The buffer for sending or receiving data. */
+    size_t len;         /*< The length of the buffer in Bytes. */
+} n25q128_data_t;
+
+/*
+ * @brief   Type definition for the configuration of an N25Q128.
+ */
+typedef struct {
+    spi_t bus;          /*< The used SPI bus */
+    spi_mode_t mode;    /*< The mode of the SPI */
+    spi_clk_t clk;      /*< The clock of the SPI */
+    gpio_t cs;          /*< The Chip-Select GPIO */
+    gpio_t write;       /*< The Write GPIO */
+    gpio_t hold;        /*< The Hold GPIO */
+} n25q128_conf_t;
+
+/*
+ * @brief   Type definition for a N25Q128 device.
+ *
+ */
+typedef struct {
+    n25q128_conf_t conf;    /*< The configuration (SPI/GPIO) of the device. */
+    n25q128_cmd_t cmd;      /*< The current instruction set for the device. */
+    n25q128_data_t data;    /*< The in/out data for the current instruction. */
+    n25q128_opt_t opt;      /*< bitwise-options, for enabling additional functions. */
+} n25q128_dev_t;
+
+/*
+ * @brief   Initialize the N25Q128 with all needed peripherals.
+ *
+ * NOTE:    To initialize a device, it is only required to assign the peripheral
+ *          parameters to the configuration type (n25q128_conf_t). The other
+ *          structures must not be modified.
+ *
+ * @param[in] *dev  A pointer to the configured device.
+ */
+int n25q128_init(n25q128_dev_t *dev);
+
+/*
+ * @brief   Reads the content of the "Read Identification"-register.
+ *
+ * The Read Identification (RDID) instruction allows to read the device
+ * identification data:
+ * > Manufacturer identification (1 byte)
+ * > Device identification (2 bytes)
+ * > A Unique ID code (UID) (17 bytes, of which 14 factory programmed upon
+ * customer request).
+ *
+ * @param[in] *dev      A pointer to the configured device.
+ * @param[out] *buf     Stores the id from the memory device.
+ * @param[in] buf_len   The length of the *buf.
+ */
+void n25q128_read_id(n25q128_dev_t *dev, uint8_t *buf, size_t len);
+
+/*
+ * @brief   Read a specific amount of data bytes from an address.
+ *
+ * @param[in] *dev      A pointer to the configured device.
+ * @param[in] address   The address to read bytes from.
+ * @param[out] *buf     Stores the bytes from the address of the device.
+ * @param[in] buf_len   The length of the *buf.
+ */
+void n25q128_read_data_bytes(n25q128_dev_t *dev, uint32_t addr, uint8_t *buf, size_t len);
```

Comment:
Are there other bytes than "data" bytes? If not, I would simply name this function `n25q128_read_bytes`.